SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialEventMessages.h

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "SpatialCommonTypes.h"
+
+#include <WorkerSDK/improbable/c_worker.h>
+
+#include "SpatialEventMessages.generated.h"
+
+static_assert(sizeof(Worker_EntityId) == sizeof(int64), "EntityId assumed 64-bit here");
+static_assert(sizeof(Worker_ComponentId) == sizeof(uint32), "ComponentId assumed 32-bit here");
+static_assert(sizeof(VirtualWorkerId) == sizeof(uint32), "VirtualWorkerId assumed 32-bit here");
+static_assert(sizeof(TraceKey) == sizeof(int32), "TraceKey assumed 32-bit here");
+static_assert(sizeof(Worker_RequestId) == sizeof(int64), "RequestId assumed 64-bit here");
+
+#define GDK_EVENT_NAMESPACE "unreal_gdk."
+
+USTRUCT()
```

Comment:
If the runtime overhead of using reflection to construct event data from these structs is reasonable then looks like a relatively nice way to do it actually.

Reply:
I think that we could encapsulate it much better with a builder though. The other problem (as well as unknown runtime overhead with doing this) is that none of it was obvious on first reading to me. Seeing UProperty everywhere tells me it's doing something clever but I don't know what, and it means the actual event logger needs to do all sorts of unreal reflection parsing.

An alternative would be to say the event logger requires well formed events and that we make other utilities that turn whatever data you give them into well formed events.

Then we can remove all the inheritance and reflection stuff here. You probably don't really need the reflection stuff anywhere in that case as you know at the call site what the data is.

Reply:
As the logger really only really seems to want strings this would seem simpler.

Reply:
E.g.
```
EventMessage Event = EventLogBuilder()
    .WithString(...)
    .WithActor(...);
```
Ultimately it's just a specialised string interpolator.